pyqt-apps/siriushla/li_va_control/details.py

```diff
@@ -0,0 +1,324 @@
+                        lbl.setStyleSheet(
+                            "font: 8pt;max-height:0.6em; min-width:2em;")
+                        lay.addWidget(lbl, pos[0], pos[1])
+                    pos[0] += 1
+                    lay.addWidget(
+                        self.getWidget(name[12:], item['widget']),
+                        pos[0], pos[1])
+                    pos[1] += 1
+                    pos[0] = 0
+
+        return wid
+
+    def setupVgcListDet(self):
+        """ Display VGC Widget List with all VGC elements """
+        wid, lay = self.getLayoutWidget("G")
+        pos = [0, 1]
+        for title, data in VGC_DETAILS.items():
+            pos[0] = 0
+            if title != 'led':
+                label = QLabel('<strong>'+title+'</strong>')
+                label.setAlignment(Qt.AlignCenter)
+                lay.addWidget(label, pos[0], pos[1], 1, 1)
+            pos[0] += 1
+            pos = self.setupAllDevices(title, data, lay, pos)
+            pos[1] += 1
+        return wid
+
+    def setupIPSListDet(self):
+        """ Display IPS Widget List with all IPS elements """
+        wid, lay = self.getLayoutWidget("V")
+        cat = "Pump"
+        config = PVS_CONFIG[cat]
+        self.devpref = config['prefix']
+        pv_range = config["iterations"]
+        for item in range(pv_range[0], pv_range[1]+1):
+            pv_name = self.devpref + self.buildIdName(item)
+            widget, lay_item = self.getLayoutWidget("H")
+            lay_item.setSpacing(0)
+            lay_item.setContentsMargins(0, 2, 0, 0)
+            lay_item.addWidget(
+                self.setWindowBtn(cat, item),
+                alignment=Qt.AlignLeft)
+            lay_item.addWidget(
+                QLabel("<strong>IPS"+str(item)+"</strong>"),
+                alignment=Qt.AlignLeft)
+            self.buildIPSInfo(pv_name, lay_item, "H")
+            lay_item.addWidget(
+                self.setupIPSControl(pv_name))
+            lay.addWidget(widget)
+        return wid
+
+    def buildList(self, cat):
+        """ Select which widget list to build """
+        if cat == "Vacuum":
+            wid = self.setupVgcListDet()
+        else:
+            wid = self.setupIPSListDet()
+        return wid
+
+    def widgetLists(self):
+        """ Build all lists """
+        wid, lay = self.getLayoutWidget("H")
+        lay.setContentsMargins(0, 0, 0, 0)
+        size = 3
+        for cat in PVS_CONFIG:
+            if cat != "Valve":
+                group = QGroupBox()
+                lay_group = QVBoxLayout()
+                lay_group.setSpacing(0)
+                lay_group.setContentsMargins(0, 0, 0, 0)
+                group.setTitle(cat)
+                group.setLayout(lay_group)
+                lay_group.addWidget(self.buildList(cat))
+                lay.addWidget(group, size)
+                size = 1
+        return wid
+
+    def _setupUi(self):
+        """."""
+        wid, lay = self.getLayoutWidget("V")
```

Comment:
maybe it is better to create only the layout here, instead of using this method, because this way the `lay` is already set to `wid`within the method, and in line 324 you set the same layout to the class, it can cause problems